rpc/server_options.go

```diff
@@ -358,6 +362,15 @@ func WithAuthHandler(forType CredentialsType, handler AuthHandler) ServerOption
 	})
 }
 
+// WithEnsureAuthedHandler returns a ServerOptions which adds custom logic for
+// the ensuring of authentication on each incoming request.
+func WithEnsureAuthedHander(eah func(ctx context.Context) (context.Context, error)) ServerOption {
```

Comment:
Had to add this to inject into `ensureAuthed`/examine its behavior. If we're nervous about making a method like this public, I can _probably_ find another way.